api/python/quilt3/data_transfer.py

```diff
@@ -444,8 +473,10 @@ def _copy_file_list_internal(file_list, message, callback):
 
     s3_client_provider = S3ClientProvider()  # Share provider across threads to reduce redundant public bucket checks
 
-    with tqdm(desc=message, total=total_size, unit='B', unit_scale=True) as progress, \
-         ThreadPoolExecutor(s3_transfer_config.max_request_concurrency) as executor:
+    # Get the appropriate executor
+    executor = _get_executor(s3_transfer_config.max_request_concurrency)
```

Comment:
You should still use `with` here, to ensure the executor gets closed. You can probably make it work using `@contextmanager`.

Reply:
Good idea!